modules/services/window-managers/i3-sway/lib/functions.nix

```diff
@@ -176,4 +155,7 @@ rec {
     "for_window ${criteriaStr criteria} ${command}";
   workspaceOutputStr = item:
     ''workspace "${item.workspace}" output ${item.output}'';
+
+  # Source: https://github.com/nix-community/home-manager/blob/master/modules/services/xidlehook.nix#L10 | Optional strings will else generate empty lines
+  notEmpty = list: filter (x: x != "" && x != null) (flatten list);
```

Comment:
Can this be moved before its usage?

Reply:
If you mean moving it to lib yeah I was not sure if I have to split it up: https://github.com/nix-community/home-manager/pull/2314#discussion_r725502626